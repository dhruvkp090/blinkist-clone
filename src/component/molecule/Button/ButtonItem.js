import Button from '@material-ui/core/Button';
import big from '../../../assets/logo/big.svg';
import small from '../../../assets/logo/small.svg';
import icons from '../../../assets/icons';
import PropTypes from 'prop-types';


const ButtonItem = (props) => {
    return (props.endIconNeeded? 
                <Button data-testid="icon-button" aria-label={props.children} role="button" 
                    endIcon = {ifOpenRenderUpArrowElseDownArrow(props)} {...props} >
                    {props.children}
                </Button> : 
                <Button role="button" {...props} style={{justifyContent:"center"}}>
                    {renderButtonWithLogoOrText(props)}
                </Button>
    );       
};

export default ButtonItem;


function ifOpenRenderUpArrowElseDownArrow(props) {
    return props.isOpen ? icons.open.src : icons.close.src;
}

function renderButtonWithLogoOrText(props) {
    return props.logoSize === "big" ?
        <img style={{ display: 'block', minWidth: "9rem" }} src={big} alt="large logo" /> :
        (renderSmallButtonOrButtonWithText());

    function renderSmallButtonOrButtonWithText() {
        return props.logoSize === "small" ?
            <img src={small} alt="small logo" /> : props.children;
    }
}

ButtonItem.propTypes = {
    isOpen: PropTypes.bool,
};