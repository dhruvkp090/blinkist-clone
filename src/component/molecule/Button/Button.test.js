import React from 'react';
import {render, screen} from '@testing-library/react';
import ButtonItem from './ButtonItem';
import '@testing-library/jest-dom';

it('Button name should equal the children prop', () => {
    const { getByText } = render(<ButtonItem children="Test" />);

    getByText("Test");
});

it('Button must have attribute end icon if endIcon needed prop is passed', () => {
    render(<ButtonItem children="Test" endIconNeeded="more" isOpen={false} />);

    screen.getByTestId("icon-button");
    
});

it('Button must have big logo if big is passed in props', () => {
    render(<ButtonItem children="Test" logoSize="big" />);

    screen.getByAltText("large logo");
    expect(screen.queryByAltText("small logo")).toBeNull();
    expect(screen.queryByTestId("icon-button")).toBeNull();
});

it('Button must have small logo if small is passed in props', () => {
    render(<ButtonItem children="Test" logoSize="small" />);

    screen.getByAltText("small logo");
    expect(screen.queryByAltText("large logo")).toBeNull();
    expect(screen.queryByTestId("icon-button")).toBeNull();
});

it('Snapshot for big logo', () => {
    const { asFragment } = render(<ButtonItem children="Test" logoSize="big" />);

    expect(asFragment()).toMatchSnapshot();
});

it('Snapshot for small logo', () => {
    const { asFragment } = render(<ButtonItem children="Test" logoSize="small" />);

    expect(asFragment()).toMatchSnapshot();
});

it('Snapshot for down arrow', () => {
    const { asFragment } = render(<ButtonItem children="Test" endIconNeeded="more" isOpen={false} />);

    expect(asFragment()).toMatchSnapshot();
});

it('Snapshot for up arrow', () => {
    const { asFragment } = render(<ButtonItem children="Test" endIconNeeded="more" isOpen={true} />);

    expect(asFragment()).toMatchSnapshot();
});

it('Normal button', () => {
    const { asFragment } = render(<ButtonItem children="Test" />);

    expect(asFragment()).toMatchSnapshot();
});