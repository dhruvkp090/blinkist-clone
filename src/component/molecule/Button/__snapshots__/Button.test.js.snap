// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normal button 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    role="button"
    style="justify-content: center;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Snapshot for big logo 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    logosize="big"
    role="button"
    style="justify-content: center;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <img
        alt="large logo"
        src="big.svg"
        style="display: block; min-width: 9rem;"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Snapshot for down arrow 1`] = `
<DocumentFragment>
  <button
    aria-label="Test"
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    data-testid="icon-button"
    endiconneeded="more"
    role="button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Snapshot for small logo 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    logosize="small"
    role="button"
    style="justify-content: center;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <img
        alt="small logo"
        src="small.svg"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Snapshot for up arrow 1`] = `
<DocumentFragment>
  <button
    aria-label="Test"
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    data-testid="icon-button"
    endiconneeded="more"
    role="button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
